eng/ci/templates/jobs/build.yml

```diff
@@ -0,0 +1,53 @@
+parameters:
+- name: pool
+  type: string
+  default: '1es-pool-azfunc'
+
+jobs:
+  - job: Build
+    displayName: 'Build'
+
+    pool:
+      name: ${{ parameters.pool }}
+      image: 1es-ubuntu-22.04
+      os: linux
+
+    variables:
+      NODE_VERSION: '20.x'
+
+    steps:
+    - checkout: self
+
+    - task: UseNode@1
+      inputs:
+        version: ${{ variables.NODE_VERSION }}
+      displayName: 'Setup Node'
+
+    - task: Npm@1
+      inputs:
+        command: 'install'
+      displayName: 'Install dependencies'
+
+    - task: Npm@1
+      inputs:
+        command: 'custom'
+        customCommand: 'run build'
+      displayName: 'Build'
+
+    - bash: |
+        cd ./tests/e2e/java8
+        mvn clean package
+        mvn azure-functions:package
+      displayName: 'Build project for java language'
+
+    - task: UseDotNet@2
+      displayName: 'Use .NET Core sdk'
+      inputs:
+        packageType: sdk
+        version: 3.1.x
+
+    - pwsh: |
```

Comment:
nit: use the [dotnet task](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/dotnet-core-cli-v2?view=azure-pipelines)

``` yml
- task: DotNetCoreCLI@2
  inputs:
    command: build
    projects: <- path to csproj here ->
    arguments: -v m
```